hazelcast/src/main/java/com/hazelcast/client/map/impl/nearcache/NearCachedClientMapProxy.java

```diff
@@ -516,7 +518,9 @@ public <R> InternalCompletableFuture<R> submitToKeyInternal(Object key,
         try {
             future = super.submitToKeyInternal(key, entryProcessor);
         } finally {
-            invalidateNearCache(key);
+            if (!(entryProcessor instanceof ReadOnly)) {
```

Comment:
If you define your EP as read-only on the client-side, but not read-only on the server side (or vice versa), you might not invalidate an entry you should (or vice versa). 

We could most probably say this is a user error but just writing it here to get more ideas on it, whether it is fine or not.

Reply:
We should ideally not invalidate if it is read only. And probably mention it in the documentation?